### PR TITLE
Query available languages

### PR DESCRIPTION
**Pull request by hackthehackerman**
Added a query to get a list of available languages for latest_modules and the count for each languages.

### CODE REVIEW

**Comment by reedstrm** [comment]
This looks good, setting up the backend retrieval. We typically merge complete features to master, however. Let's get the changes to the extras view in there  (should be a one-liner right about https://github.com/Connexions/cnx-archive/blob/master/cnxarchive/views.py#L883-L888) and update the tests, and we'll be good to go.

**Comment by hackthehackerman** [comment]
@reedstrm Sorry, I was not familiarized with the architecture. Based on the example, I assume I should add `'languages_and_counts': _get_available_languages_and_count(x),` to where you were pointing to?

**Comment by reedstrm** [comment]
yup, looks like it should work. Then run 'python setup.py test' in the top level (cnx-archive) directory. I presume some tests will fail. Update the expected output.

File: cnxarchive/views.py [hunk 1]
```diff
@@ -399,6 +399,12 @@ def _get_subject_list(cursor):
         yield subject
 
 
+def _get_available_languages_and_count(cursor):
+    """Return a list of available language and its count"""
+    cursor.execute(SQL['get-available-languages-and-count'])
+    return [i[0] for i in cursor.fetchall()]
```
Comment (karenc) [inline cnxarchive/views.py]: Is this going to return just the language without the count?

File: cnxarchive/views.py [hunk 2]
```diff
@@ -402,7 +402,7 @@ def _get_subject_list(cursor):
 def _get_available_languages_and_count(cursor):
     """Return a list of available language and its count"""
     cursor.execute(SQL['get-available-languages-and-count'])
-    return [i[0] for i in cursor.fetchall()]
+    return [i[0:2] for i in cursor.fetchall()]
```
Comment (hackthehackerman) [inline cnxarchive/views.py]: @karenc Thanks for pointing it out. I think this is fixed now.

File: cnxarchive/views.py [hunk 3]
```diff
@@ -399,6 +399,12 @@ def _get_subject_list(cursor):
         yield subject
 
 
+def _get_available_languages_and_count(cursor):
+    """Return a list of available language and its count"""
+    cursor.execute(SQL['get-available-languages-and-count'])
+    return [i[0:2] for i in cursor.fetchall()]
```
Comment (karenc) [inline cnxarchive/views.py]: You can probably just `return cursor.fetchall()` :)

**Comment by karenc** [comment]
When you're done with all the changes, if you could clean up the commits, that'll be great.

You can either use `git rebase -i master` if you're familiar with it or:

```
git reset master
git add cnxarchive/sql/get-available-languages-and-count.sql
git commit -a
git push -f origin HEAD
```

File: cnxarchive/sql/get-available-languages.sql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+-- ###
```
Comment (karenc) [inline cnxarchive/sql/get-available-languages.sql]: Are you working on some more changes that use this file?

Comment (hackthehackerman) [inline cnxarchive/sql/get-available-languages.sql]: @karenc no, I must have forgotten to delete it. Will do.

**Comment by hackthehackerman** [comment]
@karenc I can rebase and merge my commits into one. But since my commits were already pushed to this branch, I can either push --force to rewrite the history(problematic for others who has pulled this branch before) or I can delete/make a new branch with the compact commit. Which way would you like?

**Comment by karenc** [comment]
Force push is ok for pull requests I think.

**Comment by hackthehackerman** [comment]
@karenc it's ok as in I can do that. However, this pull request is to merge my branch, which lives inside the current repo, into the master branch. If anyone else has pulled the current branch then it might create problems for them later. Since nobody else is working on this branch, I guess it's ok to do that.

**Comment by karenc** [comment]
There are ways to deal with rebased branches, I wouldn't worry too much about it :p

**Comment by hackthehackerman** [comment]
@karenc ok, thanks!

**Comment by karenc** [comment]
Looks great, thank you!

**Comment by reedstrm** [comment]
looks good. Merging.

**Pull request merged**